atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -224,7 +230,7 @@ private void upgradeFromOlderInternalSchema() {
                         log.warn("Upgrading table {} to new internal Cassandra schema", tableRef);
                         tablesToUpgrade.put(tableRef, clusterSideMetadata);
                     }
-                } else if (!(tableRef.equals(AtlasDbConstants.METADATA_TABLE) || tableRef.equals(schemaMutationLockTable.getOnlyTable()))) { // only expected cases
+                } else if (!hiddenTables.isHidden(tableRef)) {
```

Comment:
Not quite related to this PR but demonstrates why this is a confusing problem.  There is a list of hidden tables inside AtlasDbConstants and the one held in HiddenTables (also it's not clear to me why HiddenTables is ever instantiated).  My recommendation would be that HiddenTables expose layers or we break it apart into the lists the correspond to the different levels of hidden, which as I currently understand it would be has follows: NO_METADATA_TABLES => (METADATA_TABLE, TIMESTAMP_TABLE), ATLASDB_HIDDEN_TABLES => (NO_METADATA_TABLES, PUNCH_TABLE, SCRUB_TABLE, TRANSACTION_TABLE), <KVS>_HIDDEN_TABLES => e.g. CASSANDRA_HIDDEN_TABLES => (ATLADB_HIDDEN_TABLES, LOCK_TABLE)